### DISCUSSION TITLE
Dynamic steps depending routes

### DISCUSSION THREAD

**Discussion by mazc26** [elrumordelaluz/reactour#544] [Q&A]
Hi, first of all, congratz on such a great lib and good api, good job there. 

My question comes because I've been trying to use the library with dynamic data depending routes, for example imagine I want to open it on "/home" and then when I go to "/details" I need to open another one with new steps but sometimes I noticed steps are like delayed, Im using the lib as a context and set the steps with the `setSteps` method it provides on each component mount where I need it but sadly it gets stuck with the previous steps

**Comment by harsz** [comment]
Any update on this , I am kinda facing same issues, navigating tour steps from one view to another in flow.

**Comment by elrumordelaluz** [comment]
Mind creating a minimal reproduction to play around?

Does [this one](https://codesandbox.io/s/reactour-tour-demo-using-react-router-dom-kujql) or [this one](https://codesandbox.io/s/tour-demo-using-react-router-dom-with-automatic-route-switching-fhdnxb?file=/src/App.js) can help?
